crates/compute/src/memory.rs

```diff
@@ -22,12 +24,49 @@ impl<T> SizedSlice for &mut [T] {
 	}
 }
 
+/// A batch of slices of the same length.
+pub struct SlicesBatch<Slice: SizedSlice> {
+	rows: Vec<Slice>,
+	row_len: usize,
+}
+
+impl<Slice: SizedSlice> SlicesBatch<Slice> {
+	/// Creates a new batch of slices with the given length
+	pub fn new(rows: Vec<Slice>, row_len: usize) -> Self {
```

Comment:
Please document the pre-condition in the docstring. Otherwise it should return an `InputValidation` error instead of asserting.